autoload/ale/completion.vim

```diff
@@ -382,7 +382,23 @@ function! ale#completion#Show(result) abort
     endif
 
     if l:source is# 'ale-import'
-        call ale#completion#HandleUserData(b:ale_completion_result[0])
+        if len(b:ale_completion_result) == 1
+            let l:index = 0
+        else
+            let l:names = map(copy(b:ale_completion_result), {k, v -> v.menu})[:9]
+            let l:userselection = ale#util#MenuPrompt(
+                        \   'Name to import',
+                        \   map(range(len(l:names)), 'string(v:val)'),
+                        \   l:names,
+                        \   1
+                        \)
+            if l:userselection is# v:null
+                return
+            else
+                let l:index = l:userselection
+            endif
+        endif
+        call ale#completion#HandleUserData(b:ale_completion_result[l:index])
```

Comment:
You can update this now I've merged the commit with the simple fix for the position. What you can probably do here is this, merging this with the above conditions.

```vim
if l:source is# 'ale-import' && len(b:ale_completion_result) == 1
    call ale#completion#HandleUserData(b:ale_completion_result[0])
    
    let l:text_changed = '' . g:ale_lint_on_text_changed

    " Check the buffer again right away, if linting is enabled.
    if g:ale_enabled
    \&& (
    \   l:text_changed is# '1'
    \   || l:text_changed is# 'always'
    \   || l:text_changed is# 'normal'
    \   || l:text_changed is# 'insert'
    \)
        call ale#Queue(0, '')
    endif
elif l:source is# 'ale-automatic' || l:source is# 'ale-manual' || l:source is# 'ale-import'
    if l:source is# 'ale-import'
        call feedkeys('i', 'n')
    endif

    " Replace completion options shortly before opening the menu.
    call s:ReplaceCompletionOptions(l:source)
    call timer_start(0, function('s:OpenCompletionMenu'))
elif l:source is# 'ale-callback'
    call b:CompleteCallback(b:ale_completion_result)
endif
```

If there are extra possible results we should be able to just enter insert mode and show the usual completion menu, so we don't have to have code for a special menu.

Reply:
Okay. I changed it as you suggested. Only thing is instead of `feedkeys('i', 'n')`, I used `feedkeys('ea', 'n')`. This fixes it so it does not add new text in the middle of the word the cursor is on.